### PR TITLE
Feature/upmcfhir 192 multiple encounters per procedure

### PR DESCRIPTION
**Pull request by au2**
#### What does this PR do?
1. Adds Encounters to supported CCDA sections. We are supporting Procedure Encounters and those are essentially identical to Encounters Section entries.
2. Both EncountersSection and EncounterSectionEntriesOptional are added; some examples are available as EncountersSectionEntriesOptional. CCDA spec list EncountersSectionEntriesOptional as part of CCD.
3. Procedure Activity Procedure encounters are not created during transformation anymore as Spec requires an ID (not full description) of the encounter that is defined in another section. Instead, the id of the encounter is noted and after ccd is transformed it is matched to a Encounter resource from Encounters section and the context of the Procedure is updated.
4. Due to 2 Procedure Encounters are not created multiple times.
5. Code that is used by Problems and Observation to create encounter are removed as those templates do not have encounters in the spec.
6. This also put some structure and code so that other identifier based future deduplication is easier.

#### Related JIRA tickets:

UPMCFHIR-192
UPMCFHIR-195
UPMCFHIR-199
UPMCFHIR-200
UPMCFHIR-208

#### How should this be manually tested?
Find a CCDA that Procedure Activity Procedure entries with encounters and make sure context field is populated for such Procedures.

#### Background/Context:
See tickets.

#### New JIRA tickets for clinical review required?
No

### CODE REVIEW

File: src/main/java/tr/com/srdc/cda2fhir/transform/entry/impl/EntryResult.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	public void addDeferredReference(IDeferredReference deferredReference) {
+		if (deferredReferences == null) {
+			deferredReferences = new ArrayList<IDeferredReference>();
+		}
+		deferredReferences.add(deferredReference);
+	}
+
+	@Override
+	public boolean hasDeferredReferences() {
```
Comment (5t33) [inline src/main/java/tr/com/srdc/cda2fhir/transform/entry/impl/EntryResult.java:32]: Seems like a duplicate of hasDeferredReference() with the addition of a null check.

**Review by 5t33 — COMMENTED** [review]
LGTM, just the one small comment. The integration tests work locally for me and it looks like it failed on jenkins due to the docker port being allocated already, so there may have been another build at the same time. Re-running the tests should pass.

File: src/main/java/tr/com/srdc/cda2fhir/transform/entry/impl/EntryResult.java (hunk 1 above)
Comment (au2) [reply src/main/java/tr/com/srdc/cda2fhir/transform/entry/impl/EntryResult.java:32]: removed

**Comment by au2** [comment]
@steven I made the change you found and tests pass now. Can you approve?

**Review by 5t33 — APPROVED** [review]

**Pull request merged**